IMOS/finaliseData.m

```diff
@@ -161,47 +161,28 @@
       end
   end
   
-  if isempty(sam.time_deployment_start)
-      sam.time_deployment_start         = [];
-      sam.time_deployment_start_origin  = [];
-  end
-  if isempty(sam.time_deployment_end)
-      sam.time_deployment_end           = [];
-      sam.time_deployment_end_origin    = [];
-  end
-  
-  % set the time coverage period from the data
-  switch mode
-      case 'profile'
-          iTime = getVar(sam.variables, 'TIME');
-          if iTime ~= 0
-              if isempty(sam.time_coverage_start),
-                  sam.time_coverage_start = sam.variables{iTime}.data(1);
-              end
-              if isempty(sam.time_coverage_end),
-                  sam.time_coverage_end   = sam.variables{iTime}.data(end);
-              end
-          else
-              if isempty(sam.time_coverage_start), sam.time_coverage_start = []; end
-              if isempty(sam.time_coverage_end),   sam.time_coverage_end   = []; end
-          end
-          
-      case 'timeSeries'
-          iTime = getVar(sam.dimensions, 'TIME');
-          if iTime ~= 0
-              if isempty(sam.time_coverage_start),
-                  sam.time_coverage_start = sam.dimensions{iTime}.data(1);
-              end
-              if isempty(sam.time_coverage_end),
-                  sam.time_coverage_end   = sam.dimensions{iTime}.data(end);
-              end
-          else
-              if isempty(sam.time_coverage_start), sam.time_coverage_start = []; end
-              if isempty(sam.time_coverage_end),   sam.time_coverage_end   = []; end
-          end
-  
+  % set the time coverage/deployment period from the data
+  iTime = getVar(sam.variables, 'TIME');
+  if iTime ~= 0
+    if isempty(sam.time_coverage_start)
+      sam.time_coverage_start = sam.variables{iTime}.data(1);
+    end
+    if isempty(sam.time_coverage_end)
+      sam.time_coverage_end   = sam.variables{iTime}.data(end);
+    end
+    if isempty(sam.time_deployment_start)
+      sam.time_deployment_start = sam.variables{iTime}.data(1);
+    end
+    if isempty(sam.time_deployment_end)
+      sam.time_deployment_end = sam.variables{iTime}.data(end);
+    end
+  else
+    if isempty(sam.time_coverage_start), sam.time_coverage_start = []; end
+    if isempty(sam.time_coverage_end), sam.time_coverage_end   = []; end
+    if isempty(sam.time_deployment_start), sam.time_deployment_start = []; end
+    if isempty(sam.time_deployment_end), sam.time_deployment_end = []; end
```

Comment:
You don't initialise the `_origin` attributes, did you check this doesn't have a negative impact?
Also, you changed the logic in comparison to previous code, are you sure this is needed? This was not needed to tackle https://github.com/aodn/imos-toolbox/issues/541

Reply:
thanks - I'll add them in the 'deployment' if block above which is where they should be.

The reason why I changed the logic from the previous code was that were duplicate checks for the same thing several times.

Reply:
@ggalibert , in fact, _origin attributes are initialized by makeNEtCDFCompliant to an empty string. FinaliseData.m overwrite this to an empty array, which did not make any sense, since they are set to strings.
Hence, if there is ever a need to append information to the origin attribute, the current functionality already handle it (non empty values are strings).

This is why lines were deleted - I will update the commit msg.

Reply:
restore previous logic - too hard to test.